tests/test_ell1h.py

```diff
@@ -12,111 +13,223 @@
+                % (p, np.nanmax(relative_diff).value)
+            )
+            if p in ["EPS1DOT", "EPS1"]:
+                tol = 0.05
             else:
-                continue
-
-    def test_J0613_H4(self):
-        log = logging.getLogger("TestJ0613.fit_tests")
-        f = ff.GLSFitter(self.toasJ0613, self.modelJ0613)
-        f.fit_toas()
-        f.model.free_params = ("H3", "H4")
-        for pn, p in f.model.get_params_dict("free", "quantity").items():
-            op = getattr(f.model_init, pn)
-            diff = np.abs(p.value - op.value)
-            sigma = diff / op.uncertainty_value
-            # Fit th
-            assert (
-                sigma < 0.7
-            ), "refit %s is %lf sigma different from original value" % (pn, sigma)
-
-    def test_J0613_STIG(self):
-        log = logging.getLogger("TestJ0613.fit_tests_stig")
-        f = ff.GLSFitter(self.toasJ0613, self.modelJ0613_STIG)
-        f.fit_toas()
-        f.model.free_params = ("H3", "STIGMA")
-        for pn, p in f.get_params_dict("free", "quantity").items():
-            op = getattr(f.model_init, pn)
-            diff = np.abs(p.value - op.value)
-            sigma = diff / op.uncertainty_value
-            # Fit th
-            assert (
-                sigma < 0.7
-            ), "refit %s is %lf sigma different from original value" % (pn, sigma)
-
-
-if __name__ == "__main__":
-    pass
+                tol = 1e-3
+            log.debug(
+                "derivative relative diff for %s, %lf"
+                % ("d_delay_d_" + p, np.nanmax(relative_diff).value)
+            )
+            assert np.nanmax(relative_diff) < tol, msg
+        else:
+            continue
+
+
+def test_J0613_H4(toasJ0613, modelJ0613):
+    log = logging.getLogger("TestJ0613.fit_tests")
+    f = ff.GLSFitter(toasJ0613, modelJ0613)
+    f.fit_toas()
+    f.model.free_params = ("H3", "H4")
+    for pn, p in f.model.get_params_dict("free", "quantity").items():
+        op = getattr(f.model_init, pn)
+        diff = np.abs(p.value - op.value)
+        sigma = diff / op.uncertainty_value
+        # Fit th
+        assert sigma < 0.7, "refit %s is %lf sigma different from original value" % (
+            pn,
+            sigma,
+        )
+
+
+def test_J0613_STIG(toasJ0613, modelJ0613_STIG):
+    log = logging.getLogger("TestJ0613.fit_tests_stig")
+    f = ff.GLSFitter(toasJ0613, modelJ0613_STIG)
+    f.fit_toas()
+    f.model.free_params = ("H3", "STIGMA")
+    for pn, p in f.get_params_dict("free", "quantity").items():
+        op = getattr(f.model_init, pn)
+        diff = np.abs(p.value - op.value)
+        sigma = diff / op.uncertainty_value
+        # Fit th
+        assert sigma < 0.7, "refit %s is %lf sigma different from original value" % (
+            pn,
+            sigma,
+        )
+
+
+def test_no_H3_H4(toasJ0613):
+    """ Test no H3 and H4 in model.
+    """
+    no_H3_H4 = simple_par.replace("H4 2.0262048E-7  1       1.1276173E-7", "")
```

Comment:
This will silently stop working if the value or uncertainty change in any digit, or the spacing changes. Perhaps a regular expression to replace r"^H4 .*$" with nothing would be more reliable?

Reply:
Since the simple_par is a defined string in the beginning. I don't expect, it gets changed.